### ISSUE TITLE
Flake: TestConfigureTCPRouteBackingDestinationsWithMultiServices

### ISSUE THREAD

**Issue by sheidkamp** [solo-io/gloo#10366]
### Which tests failed?

TestConfigureTCPRouteBackingDestinationsWithMultiServices

https://github.com/solo-io/gloo/actions/runs/11886116108/job/33116812397

```
=== NAME  TestK8sGateway
    curl.go:58: 
        Timed out after 20.000s.
        failed to get expected response
        The function passed to Eventually failed at /home/runner/work/gloo/gloo/test/kubernetes/testutils/assertions/curl.go:47 with:
        Unexpected error:
            <*errors.withStack | 0xc001e96708>: 
            exit status 7
            {
                error: <*exec.ExitError | 0xc0069b8480>{
                    ProcessState: {
                        pid: 146349,
                        status: 1792,
                        rusage: {
                            Utime: {Sec: 0, Usec: 50815},
                            Stime: {Sec: 0, Usec: 15879},
                            Maxrss: 1424212,
                            Ixrss: 0,
                            Idrss: 0,
                            Isrss: 0,
                            Minflt: 1698,
                            Majflt: 0,
                            Nswap: 0,
                            Inblock: 0,
                            Oublock: 0,
                            Msgsnd: 0,
                            Msgrcv: 0,
                            Nsignals: 0,
                            Nvcsw: 354,
                            Nivcsw: 45,
                        },
                    },
                    Stderr: nil,
                },
                stack: [0x3e688ae, 0x5456c66, 0x545686a, 0x545645c, 0x82bdf8b, 0x2baf07d, 0x2bae056, 0x3e7ab4a, 0x3e7c90c, 0x3e78128, 0x82bdd7a, 0x82be426, 0x900745a, 0x2baf07d, 0x2bae056, 0x3e4b9f2, 0x2c55447, 0x2acc481],
            }
        occurred
tcproute.gateway.networking.k8s.io "tcp-app-1" deleted
tcproute.gateway.networking.k8s.io "tcp-app-2" deleted
service "foo" deleted
service "bar" deleted
deployment.apps "backend-1" deleted
deployment.apps "backend-2" deleted
gateway.gateway.networking.k8s.io "tcp-gateway" deleted
namespace "curl" deleted
pod "curl" deleted
=== NAME  TestK8sGateway/TCPRouteServices/TestConfigureTCPRouteBackingDestinationsWithMultiServices
    testing.go:1577: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
```

**Comment by sheidkamp** [comment]
1.18 PR - https://github.com/solo-io/gloo/actions/runs/11974004519/job/33387161746?pr=10388

**Comment by sam-heilbron** [comment]
PR: https://github.com/solo-io/gloo/pull/10406

**Comment by sam-heilbron** [comment]
@ryanrolds @danehans It looks like this test flake is addressed by https://github.com/solo-io/gloo/pull/10418 a PR that @danehans opened. Let's work to get that merged and then close out this issue if it addresses it.

**Comment by danehans** [comment]
https://github.com/solo-io/gloo/issues/10397 is a dupe.

**Comment by danehans** [comment]
Fixed by #10418
